app/models/concerns/account_statuses_search.rb

```diff
@@ -26,19 +26,11 @@ def enqueue_remove_from_public_statuses_index
   def add_to_public_statuses_index!
     return unless Chewy.enabled?
 
-    batch_size = 1000
-    offset = 0
-
-    loop do
-      batch = Status.where(account_id: id).offset(offset).limit(batch_size)
-
-      break if batch.empty?
-
+    Status.where(account_id: id).find_in_batches(batch_size: 1_000) do |batch|
```

Comment:
I believe  the `import` call will combine the `query` parameter with the scope provided in the index definition, which will filter by `where(visibility: :public)`, however, I feel like we could iterate over fewer batches if we added that condition upfront here as well.